File: backend/models/board.go
package models

type CardResponse struct {
	ID          string `json:"id"`
	Title       string `json:"title"`
	Description string `json:"description"`
	Position    int    `json:"position"`
}

type ListResponse struct {
	ID       string         `json:"id"`
	Name     string         `json:"name"`
	Position int            `json:"position"`
	Cards    []CardResponse `json:"cards"`
}

type BoardFullResponse struct {
	ID    string         `json:"id"`
	Name  string         `json:"name"`
	Lists []ListResponse `json:"lists"`
}

File: frontend/src/app/types/board.ts
export interface Board {
  id: string;
  name: string;
  ownerId: string;
  owner: {
    username: string;
    email: string;
  };
  members: Member[];
}

export interface User {
  ID: string;
  Username: string;
  Email: string;
}

export interface Member {
  id: string;
  username: string;
  email: string;
  isMember: boolean;
}

export interface CardinBoard {
  id: string;
  title: string;
  description: string;
  position: number;
}

export interface ListinBoard {
  id: string;
  name: string;
  position: number;
  cards: CardinBoard[];
}

export interface BoardDetail {
  id: string;
  name: string;
  lists: ListinBoard[];
}